cairo/src/contracts/token/extensions/hyp_erc721_URI_storage.cairo

```diff
@@ -1,48 +1,226 @@
+        fn upgrade(ref self: ContractState, new_class_hash: starknet::ClassHash) {
+            self.ownable.assert_only_owner();
+            self.upgradeable.upgrade(new_class_hash);
         }
+    }
 
-        fn token_uri(self: @ContractState, token_id: u256) -> u256 {
-            0
-        }
+    // #[generate_trait]
+    // impl InternalImpl of InternalTrait {
+    //     fn transfer_from_sender(ref self: ContractState, token_id: u256) -> u256 {
+    //         self.hyp_erc721.transfer_from_sender(token_id);
+    //         token_id
+    //     }
 
-        fn supports_interface(self: @ContractState, interface_id: u256) -> bool {
-            false
-        }
-    }
+    //     fn transfer_to(ref self: ContractState, recipient: ContractAddress, token_id: u256, token_uri: u256) {
+    //         self.hyp_erc721.transfer_to(recipient, token_id);
+    //     }
 
-    #[generate_trait]
-    impl InternalImpl of InternalTrait {
-        fn transfer_from_sender(ref self: ContractState, token_id: u256) -> u256 {
-            0
-        }
+    //     fn before_token_transfer(
+    //         ref self: ContractState, from: u256, to: ContractAddress, token_id: u256, batch_size: u256
+    //     ) {
+    //         self.erc721.before_update(to, token_id, Zero::zero());
+    //     }
+    // }
 
-        fn transfer_to(ref self: ContractState, recipient: u256, token_id: u256, token_uri: u256) {}
+    // would be extended when erc721_enumerable is imported
+    impl ERC721HooksImpl of ERC721HooksTrait<ContractState> {
+        fn before_update(
+            ref self: ERC721Component::ComponentState<ContractState>,
+            to: ContractAddress,
+            token_id: u256,
+            auth: ContractAddress
+        ) {}
 
-        fn before_token_transfer(
-            ref self: ContractState, from: u256, to: u256, token_id: u256, batch_size: u256
+        fn after_update(
+            ref self: ERC721Component::ComponentState<ContractState>,
+            to: ContractAddress,
+            token_id: u256,
+            auth: ContractAddress
         ) {}
+    }
 
-        fn burn(ref self: ContractState, token_id: u256) {}
+
+    impl TokenRouterHooksImpl of TokenRouterHooksTrait<ContractState> {
+        fn transfer_from_sender_hook(
+            ref self: TokenRouterComponent::ComponentState<ContractState>,
+            amount_or_id: u256
+        ) -> Bytes {
+            let contract_state = TokenRouterComponent::HasComponent::get_contract(@self);
+            let token_owner = contract_state.erc721.owner_of(amount_or_id);
+            assert!(
+                token_owner == get_caller_address(),
+                "Caller is not owner of token"
+            );
+
+            let mut contract_state = TokenRouterComponent::HasComponent::get_contract_mut(ref self);
+            contract_state.erc721.burn(amount_or_id);
+
+            BytesTrait::new_empty()
+        }
+
+        fn transfer_to_hook(
```

Comment:
This also does not look right, you can just call the self.hyp_erc721 implementation without rewriting the code and it shuold also set the token_uri

We need to implement [this](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/cae60c595b37b1e7ed7dd50ad0257387ec07c0cf/contracts/token/ERC721/extensions/ERC721URIStorage.sol#L14) extension